website/docs/r/virtual_machine_configuration_policy_assignment.html.markdown

```diff
@@ -0,0 +1,142 @@
+  location            = azurerm_resource_group.example.location
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.example.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_windows_virtual_machine" "example" {
+  name                = "examplevm"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_F2"
+  admin_username      = "adminuser"
+  admin_password      = "P@$$w0rd1234!"
+  network_interface_ids = [
+    azurerm_network_interface.example.id,
+  ]
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2016-Datacenter"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_virtual_machine_configuration_policy_assignment" "example" {
+  name               = "example-gca"
+  location           = azurerm_linux_virtual_machine.example.id
+  virtual_machine_id = azurerm_linux_virtual_machine.example.id
+  guest_configuration {
+    name    = "WhitelistedApplication"
+    version = "1.*"
+
+    parameter {
+      name  = "[InstalledApplication]bwhitelistedapp;Name"
+      value = "NotePad,sql"
+    }
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Virtual Machine Configuration Policy Assignment. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure location where the Virtual Machine Configuration Policy Assignment should exist. Changing this forces a new resource to be created.
+
+* `virtual_machine_id` - (Required) The resource ID of the Virtual Machine which this Guest Configuration Assignment should apply to. Changing this forces a new resource to be created.
+
+---
+
+* `guest_configuration` - (Required)  A `guest_configuration` block as defined below.
+
+---
+
+An `guest_configuration` block supports the following:
+
+* `name` - (Required) The name of the Guest Configuration that will be assigned in this Guest Configuration Assignment.
+
+* `parameter` - (Optional) One or more `parameter` blocks as defined below.
+
+* `version` - (Optional) The version of the Guest Configuration that will be assigned in this Guest Configuration Assignment.
+
+---
+
+An `parameter` block supports the following:
+
+* `name` - (Required) The name of the configuration parameter.
+
+* `value` - (Required) The value of the configuration parameter.
+
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the Virtual Machine Configuration Policy Assignment.
+
+* `compliance_status` - A value indicating compliance status of the machine for the assigned Guest Configuration.
```

Comment:
what does compliance mean? the named parameter is the value specified?

Reply:
This means the progress that policy is applying to this virtual machine - it might be `Pending` which means the policy validation does not start or `Compliant` which means the validation has finished or `Non-Compliant` which means the VM fails the validation. 
But this value could change over time as the validation will be performed in the backend. Therefore I am going to remove this attribute from terraform resource